### ISSUE TITLE
Selecting entry from Ctrl+R doesn't place it on the cmd line if $HISTSIZE is set

### ISSUE THREAD

**Issue by timothyfroehlich** [junegunn/fzf#4095]
### Checklist

- [X] I have read through the manual page (`man fzf`)
- [X] I have searched through the existing issues
- [X] For bug reports, I have checked if the bug is reproducible in the latest version of fzf

### Output of `fzf --version`

0.56.2 (brew)

### OS

- [ ] Linux
- [X] macOS
- [ ] Windows
- [ ] Etc.

### Shell

- [ ] bash
- [X] zsh
- [ ] fish

### Problem / Steps to reproduce

https://github.com/user-attachments/assets/fcee678b-95e0-4eca-91b0-5345fce400bc

This just started happening a couple days ago without any changes to my .zshrc. If I press ctrl+R and select an entry by pressing enter I get a fresh cmd line. I went through my .zshrc until I found the line that caused the problem, which seems to be `HISTSIZE=1000`.

Here's the portion of my .zshrc that's related to history:
```
HISTFILE=~/.histfile
HISTSIZE=1000
SAVEHIST=$HISTSIZE
setopt appendhistory
setopt share_history #share history between multiple instances of zsh
[ -f ~/.fzf.zsh ] && source ~/.fzf.zsh
```

Unsetting HISTSIZE or SAVEHIST doesn't seem to help, only opening a new shell without HISTSIZE set. 

Please let me know if there's anything else I can add to help debug.

**Comment by LangLangBart** [comment]
Should be fixed for the dev version. Ref: #4091

The current workaround until a new release is published is: 

```bash
source <(fzf --zsh | sed 's/fc -RI/true/')
```

Sorry for the inconvenience.

**Comment by junegunn** [comment]
Should be fixed in 0.56.3.
